src/store/pool/hooks.ts

```diff
@@ -355,3 +357,63 @@ export function useAPYs() {
 
   return apys;
 }
+
+export enum PairState {
+  LOADING,
+  NOT_EXISTS,
+  EXISTS,
+  INVALID,
+}
+
+export function useV2Pairs(currencies: [Currency | undefined, Currency | undefined][]): [PairState, Pair | null][] {
+  const [reserves, setReserves] = useState<({ reserve0: string; reserve1: string } | number | undefined)[]>([]);
+
+  const tokens = useMemo(() => {
+    setReserves(Array(currencies.length).fill(PairState.LOADING));
```

Comment:
What about moving the logic of initializing the array in the `fetchReserves` function?
The code looks strange to me.

Reply:
![image](https://user-images.githubusercontent.com/50083174/144618028-6ead38e7-852d-4a89-8869-502ee2a68461.png)

Reply:
Agreed!